### PR TITLE
Feat/lock app biometrics

### PR DESCRIPTION
**Pull request by juriczech**
## Related Issue

Resolve 

## Screenshots:

https://github.com/trezor/trezor-suite/assets/36101761/40386dbf-3997-4faa-b217-3885af61f3ed

https://github.com/trezor/trezor-suite/assets/36101761/fe46d812-a75a-420d-8c92-67d715dd6d82

### CODE REVIEW

**Comment by tomasklim** [comment]
Is it possible to require one last Face Id to be able to disable biometrics?

**Comment by juriczech** [comment]
> Is it possible to require one last Face Id to be able to disable biometrics? 

When you have biometrics enabled, you will have to authenticate to enter the app and edit settings. Disabling won't ask you to authenticate again since you're already in the app after verification.


But yes, technically it would be possible. Just unnecessary IMO.

File: suite-native/biometrics/src/components/AuthenticatorProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+type AuthenticatorProviderProps = {
+    children: ReactNode;
+};
+
+export const AuthenticatorProvider = ({ children }: AuthenticatorProviderProps) => {
+    const { isBiometricsOptionEnabled } = useIsBiometricsEnabled();
+    const { isUserAuthenticated, setIsUserAuthenticated } = useIsUserAuthenticated();
+
+    useEffect(() => {
+        const authenticate = async () => {
```
Comment (Nodonisko) [inline suite-native/biometrics/src/components/AuthenticatorProvider.tsx]: I would move this outside of useEffect using useCallback

File: suite-native/biometrics/src/components/AuthenticatorProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            if (Platform.OS === 'android') {
+                // Cancel any previous authentication attempt.
+                await LocalAuthentication.cancelAuthenticate();
+            }
+
+            const result = await LocalAuthentication.authenticateAsync();
+            setIsUserAuthenticated(result.success);
+        };
+
+        if (isBiometricsOptionEnabled && !isUserAuthenticated) {
```
Comment (Nodonisko) [inline suite-native/biometrics/src/components/AuthenticatorProvider.tsx:28]: I would split this if and subsription lower to two separeted useEffects.

**Review by Nodonisko — CHANGES_REQUESTED** [review]

File: suite-native/biometrics/src/components/AuthenticatorProvider.tsx (hunk 2 above)
Comment (juriczech) [reply suite-native/biometrics/src/components/AuthenticatorProvider.tsx:28]: https://github.com/trezor/trezor-suite/pull/8540/commits/b0340b2b1efdda51d31bc3a8bdd0dd4c36cbdda6

File: suite-native/biometrics/src/components/AuthenticatorProvider.tsx (hunk 1 above)
Comment (juriczech) [reply suite-native/biometrics/src/components/AuthenticatorProvider.tsx]: https://github.com/trezor/trezor-suite/pull/8540/commits/b0340b2b1efdda51d31bc3a8bdd0dd4c36cbdda6

**Review by Nodonisko — APPROVED** [review]

**Pull request merged**